File: corporate-hotel-booking/src/hotel/hotel_service.py
from src.hotel.hotel import Hotel, RoomType, Room
from src.hotel.hotel_repository import HotelRepository


class HotelService:
    def __init__(self, hotel_repository: HotelRepository):
        self.__hotel_repository = hotel_repository

    def add_hotel(self, hotel_id: str, hotel_name: str):
        self.__hotel_repository.add_hotel(hotel=Hotel(id=hotel_id, name=hotel_name))

    def set_room(self, hotel_id: str, room_number: int, room_type: RoomType):
        hotel = self.__hotel_repository.find_hotel_by_id(hotel_id=hotel_id)
        hotel.add_room(room=Room(number=room_number, type=room_type))
        self.__hotel_repository.save(hotel=hotel)

    def find_hotel_by(self, hotel_id: str):
        return self.__hotel_repository.find_hotel_by_id(hotel_id=hotel_id)

File: corporate-hotel-booking-java/src/main/java/com/github/ambersariya/hotel/HotelService.java
package com.github.ambersariya.hotel;

import java.util.List;

public class HotelService {

    private static final List<RoomType> SUPPORTED_ROOM_TYPES = List.of(RoomType.STANDARD, RoomType.MASTER_SUITE);
    private final HotelRepository hotelRepository;

    public HotelService(HotelRepository hotelRepository) {
        this.hotelRepository = hotelRepository;
    }

    public void addHotel(int hotelId, String hotelName) {
        if (hotelRepository.findHotelBy(hotelId) != null) {
            throw new HotelAlreadyExists("Hotel already exists");
        }

        var hotel = new Hotel(hotelId, hotelName, SUPPORTED_ROOM_TYPES);
        hotelRepository.saveHotel(hotel);
    }

    public void setRoom(int hotelId, int roomNumber, RoomType roomType) {
        var hotel = hotelRepository.findHotelBy(hotelId);
        if (hotel == null) {
            throw new HotelNotFound("Hotel does not exist");
        }
        hotel.setRoom(roomNumber, roomType);
        hotelRepository.saveHotel(hotel);
    }

    public Hotel findHotelBy(int hotelId) {
        return hotelRepository.findHotelBy(hotelId);
    }
}
